server/src/main/java/org/opensearch/action/search/SearchRequestStats.java

```diff
@@ -0,0 +1,204 @@
+        return totalStats.canMatchTotal.count();
+    }
+
+    public long getQueryMetric() {
+        return totalStats.queryMetric.sum();
+    }
+
+    public long getQueryCurrent() {
+        return totalStats.queryCurrent.count();
+    }
+
+    public long getQueryTotal() {
+        return totalStats.queryTotal.count();
+    }
+
+    public long getFetchMetric() {
+        return totalStats.fetchMetric.sum();
+    }
+
+    public long getFetchCurrent() {
+        return totalStats.fetchCurrent.count();
+    }
+
+    public long getFetchTotal() {
+        return totalStats.fetchTotal.count();
+    }
+
+    public long getExpandSearchMetric() {
+        return totalStats.expandSearchMetric.sum();
+    }
+
+    public long getExpandSearchCurrent() {
+        return totalStats.expandSearchCurrent.count();
+    }
+
+    public long getExpandSearchTotal() {
+        return totalStats.expandSearchTotal.count();
+    }
+
+    private void computeStats(Consumer<StatsHolder> consumer) {
+        consumer.accept(totalStats);
+    }
+
+    @Override
+    public void onDFSPreQueryPhaseStart(SearchPhaseContext context) {
+        computeStats(statsHolder -> { statsHolder.dfsPreQueryCurrent.inc(); });
+    }
+
+    @Override
+    public void onDFSPreQueryPhaseEnd(SearchPhaseContext context, long tookTime) {
+        computeStats(statsHolder -> {
+            totalStats.dfsPreQueryCurrent.dec();
+            totalStats.dfsPreQueryTotal.inc();
+            totalStats.dfsPreQueryMetric.inc(tookTime);
+        });
+    }
+
+    @Override
+    public void onDFSPreQueryPhaseFailure(SearchPhaseContext context) {
+        computeStats(statsHolder -> { statsHolder.dfsPreQueryCurrent.dec(); });
+    }
+
+    @Override
+    public void onCanMatchPhaseStart(SearchPhaseContext context) {
+        computeStats(statsHolder -> { statsHolder.canMatchCurrent.inc(); });
+    }
+
+    @Override
+    public void onCanMatchPhaseEnd(SearchPhaseContext context, long tookTime) {
+        computeStats(statsHolder -> {
+            totalStats.canMatchCurrent.dec();
+            totalStats.canMatchTotal.inc();
+            totalStats.canMatchMetric.inc(tookTime);
+        });
+    }
+
+    @Override
+    public void onCanMatchPhaseFailure(SearchPhaseContext context) {
+        computeStats(statsHolder -> { statsHolder.canMatchCurrent.dec(); });
+    }
+
+    @Override
+    public void onQueryPhaseStart(SearchPhaseContext context) {
+        computeStats(statsHolder -> { statsHolder.queryCurrent.inc(); });
+    }
+
+    @Override
+    public void onQueryPhaseEnd(SearchPhaseContext context, long tookTime) {
+        computeStats(statsHolder -> {
+            totalStats.queryCurrent.dec();
+            totalStats.queryTotal.inc();
+            totalStats.queryMetric.inc(tookTime);
+        });
+    }
+
+    @Override
+    public void onQueryPhaseFailure(SearchPhaseContext context) {
+        computeStats(statsHolder -> { statsHolder.queryCurrent.dec(); });
```

Comment:
Why don't we count took time of failed requests?

Reply:
Me and Sagar have decided that doesn't make sense for this feature.